recipes/liburing/all/conanfile.py

```diff
@@ -31,39 +31,42 @@ class LiburingConan(ConanFile):
 
     _autotools = None
 
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
 
-        self._autotools = AutoToolsBuildEnvironment(self)
-        self._autotools.configure(configure_dir=self._source_subfolder)
-        return self._autotools
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
 
-    @property
-    def _source_subfolder(self):
-        return os.path.join(self.source_folder, "source_subfolder")
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("liburing is supported only on linux")
+
+        # FIXME: use kernel version of build/host machine. kernel version should be encoded in profile
+        linux_kernel_version = re.match("([0-9.]+)", platform.release()).group(1)
+        if tools.Version(linux_kernel_version) < "5.1":
+            raise ConanInvalidConfiguration("This linux kernel version does not support io uring")
```

Comment:
Instead of checking the kernel version of the build machine, would adding the conan `linux-headers-generic` package as a requirement work too?

Reply:
Excellent idea!

Reply:
See #7208